client/state/initial-state.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+import localforage from 'localforage';
+import msgpack from 'msgpack-lite';
+
+/**
+ * Internal dependencies
+ */
+import { createReduxStore, reducer } from 'state';
+import { SERIALIZE, DESERIALIZE } from 'state/action-types'
+
+/**
+ * Module variables
+ */
+const debug = debugModule( 'calypso:state' );
+
+const localforageConfig = {
+	name: 'calypso',
+	storeName: 'calypso_store',
+	description: 'Calypso Storage',
+	driver: localforage.INDEXDB
+};
+
+function serialize( state ) {
+	const unencodedState = reducer( state, { type: SERIALIZE } );
+	return msgpack.encode( unencodedState );
+}
+
+function deserialize( encodedState ) {
+	const state = msgpack.decode( encodedState );
+	debug( 'decoded state', state );
+	return reducer( state, { type: DESERIALIZE } );
+}
+
+function loadInitialState( initialState ) {
+	debug( 'loading initial state' );
+	if ( initialState === null ) {
+		throw new Error( 'no initial state found in localforage' );
+	}
+	return createReduxStore( deserialize( initialState ) );
```

Comment:
Nothing just yet. If we're only concerned with persisting state for logged-in users, and bootstrapping state for logged-out users — you can ignore me ;)